diff --git a/build/config/clang/BUILD.gn b/build/config/clang/BUILD.gn
index 3b561e9b0c0a9..2bddbd82de21a 100644
--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -166,8 +166,9 @@ template("clang_lib") {
       _prefix = "lib"
       _suffix = ""
       _ext = "a"
+      _libprefix = ""
 
-      _clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"
+      _clang_lib_dir = "$clang_base_path/lib${_libprefix}/clang/$clang_version/lib"
       if (is_win) {
         _dir = "windows"
         _prefix = ""
@@ -197,7 +198,33 @@ template("clang_lib") {
         } else {
           assert(false)  # Unhandled cpu type
         }
-      } else if (is_linux || is_chromeos) {
+      } else if (is_linux) {
+        if (current_cpu == "x64") {
+          _dir = "x86_64-redhat-linux-gnu"
+          _suffix ="-x86_64"
+        } else if (current_cpu == "x86") {
+          _dir = "i386-redhat-linux-gnu"
+          _suffix = "-i386"
+        } else if (current_cpu == "arm64") {
+          _dir = "aarch64-redhat-linux-gnu"
+          _suffix = "-aarch64"
+        } else if (current_cpu == "arm") {
+          _dir = "armhf-redhat-linux-gnu"
+          _suffix = "-armhf"
+        } else if (current_cpu == "ppc64") {
+          _dir = "ppc64le-redhat-linux-gnu"
+          _suffix = "-ppc64le"
+        } else {
+          assert(false)  # Unhandled cpu type
+        }
+        # different clang lib dir in fedora/epel
+        if (clang_version == "17" || clang_version == "18" || clang_version == "19" || clang_version == "20") {
+          _suffix = ""
+        } else if (clang_version == "14" || clang_version == "15" || clang_version == "16") {
+          _libprefix = "64"
+          _dir = ""
+        }
+      } else if (is_chromeos) {
         if (current_cpu == "x64") {
           _dir = "x86_64-unknown-linux-gnu"
         } else if (current_cpu == "x86") {
diff --git a/build/rust/rust_bindgen.gni b/build/rust/rust_bindgen.gni
index 5c809c6932c84..574d04c6b6c52 100644
--- a/build/rust/rust_bindgen.gni
+++ b/build/rust/rust_bindgen.gni
@@ -19,11 +19,11 @@ if (host_os == "win") {
 
 # On Windows, the libclang.dll is beside the bindgen.exe, otherwise it is in
 # ../lib.
-_libclang_path = rust_bindgen_root
+_libclang_path = clang_base_path
 if (host_os == "win") {
   _libclang_path += "/bin"
 } else {
-  _libclang_path += "/lib"
+  _libclang_path += "/lib64"
 }
 
 # Template to build Rust/C bindings with bindgen.
diff --git a/build/rust/rust_bindgen_generator.gni b/build/rust/rust_bindgen_generator.gni
index c91916be93ba5..3377c0b5e0b56 100644
--- a/build/rust/rust_bindgen_generator.gni
+++ b/build/rust/rust_bindgen_generator.gni
@@ -18,11 +18,11 @@ if (host_os == "win") {
 
 # On Windows, the libclang.dll is beside the bindgen.exe, otherwise it is in
 # ../lib.
-_libclang_path = rust_bindgen_root
+_libclang_path = clang_base_path
 if (host_os == "win") {
   _libclang_path += "/bin"
 } else {
-  _libclang_path += "/lib"
+  _libclang_path += "/lib64"
 }
 
 # Template to build Rust/C bindings with bindgen.
